### ISSUE TITLE
[incubator/zookeeper] /var/lib/zookeeper/data/myid file is missing

### ISSUE THREAD

**Issue by vasu1124** [helm/charts#869]
I deployed the current zookeeper chart, left all values as is. Except for StorageClass, which I adapted to my local provisioner.
It seems that in my setup mount under /var/lib/zookeeper is created for root:root with drwxr-xr-x  permissions, so zookeeper (1000) can't write. I am inclined to believe that this must be the same for other provisioners as well ...
Could an init container with root rights do the setup instead?

`2017-03-28T16:49:06.590215000Z 2017-03-28 16:49:06,589 [myid:] - ERROR [main:QuorumPeerMain@85] - Invalid config, exiting abnormally
2017-03-28T16:49:06.590516000Z org.apache.zookeeper.server.quorum.QuorumPeerConfig$ConfigException: Error processing /usr/bin/../etc/zookeeper/zoo.cfg
2017-03-28T16:49:06.590806000Z 	at org.apache.zookeeper.server.quorum.QuorumPeerConfig.parse(QuorumPeerConfig.java:144)
2017-03-28T16:49:06.591114000Z Invalid config, exiting abnormally
2017-03-28T16:49:06.591414000Z 	at org.apache.zookeeper.server.quorum.QuorumPeerMain.initializeAndRun(QuorumPeerMain.java:101)
2017-03-28T16:49:06.591711000Z 	at org.apache.zookeeper.server.quorum.QuorumPeerMain.main(QuorumPeerMain.java:78)
2017-03-28T16:49:06.591963000Z Caused by: java.lang.IllegalArgumentException: /var/lib/zookeeper/data/myid file is missing
2017-03-28T16:49:06.592271000Z 	at org.apache.zookeeper.server.quorum.QuorumPeerConfig.parseProperties(QuorumPeerConfig.java:362)
2017-03-28T16:49:06.592537000Z 	at org.apache.zookeeper.server.quorum.QuorumPeerConfig.parse(QuorumPeerConfig.java:140)
2017-03-28T16:49:06.592799000Z 	... 2 more`

**Comment by kow3ns** [comment]
Did you change the following?
```yaml
securityContext:
        runAsUser: 1000
        fsGroup: 1000
```
The [security context](https://kubernetes.io/docs/concepts/policy/security-context/) ensures the correct permissions on the volume. What provisioner are you using, and does it support ownership management?

**Comment by vasu1124** [comment]
Hi,

no I did not change the securityContext.

Thanks for pointing me to the right direction. I used the gluster provisioner kubernetes.io/glusterfs with the controller[Heketi](https://github.com/heketi/heketi). I would have expected that the underlay with gluster supports ownership management. At least, I worked around by adjusting with a local mount & chmod 777 manually.

Anyway, I tried the example on AWS and the mounts came up with root:1000 and drwxrwsr-x.

Thanks!
